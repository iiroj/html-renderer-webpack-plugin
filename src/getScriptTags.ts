export default (publichPath: string = "/", files: Array<string>) => {
  const prefix = publichPath.endsWith("/") ? publichPath : publichPath + "/";

  return files
    .map(
      (file: string) =>
        `<script defer type="text/javascript" src="${prefix}${file}"></script>`
    )
    .join("\n");
};
