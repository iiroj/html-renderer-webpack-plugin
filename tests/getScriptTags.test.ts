import getScriptTags from "../src/getScriptTags";

const js = ["5749aae7.js", "2ff13846.js"];

describe("getScriptTags", () => {
  it("should generate relative script tags", () => {
    const scriptTags = getScriptTags(js);
    expect(scriptTags).toMatchSnapshot();
  });

  it("should generate absolute script tags 1", () => {
    const scriptTags = getScriptTags(js, "/");
    expect(scriptTags).toMatchSnapshot();
  });

  it("should generate absolute script tags 2", () => {
    const scriptTags = getScriptTags(js, "/foo/");
    expect(scriptTags).toMatchSnapshot();
  });

  it("should append / to path if needed", () => {
    const scriptTags = getScriptTags(js, "/bar");
    expect(scriptTags).toMatchSnapshot();
  });
});
