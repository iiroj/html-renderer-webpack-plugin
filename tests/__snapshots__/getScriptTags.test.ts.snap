// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getScriptTags should append / to path if needed 1`] = `
"<script defer type=\\"text/javascript\\" src=\\"/bar/5749aae7.js\\"></script>
<script defer type=\\"text/javascript\\" src=\\"/bar/2ff13846.js\\"></script>"
`;

exports[`getScriptTags should generate absolute script tags 1 1`] = `
"<script defer type=\\"text/javascript\\" src=\\"/5749aae7.js\\"></script>
<script defer type=\\"text/javascript\\" src=\\"/2ff13846.js\\"></script>"
`;

exports[`getScriptTags should generate absolute script tags 2 1`] = `
"<script defer type=\\"text/javascript\\" src=\\"/foo/5749aae7.js\\"></script>
<script defer type=\\"text/javascript\\" src=\\"/foo/2ff13846.js\\"></script>"
`;

exports[`getScriptTags should generate relative script tags 1`] = `
"<script defer type=\\"text/javascript\\" src=\\"/5749aae7.js\\"></script>
<script defer type=\\"text/javascript\\" src=\\"/2ff13846.js\\"></script>"
`;
